airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -237,6 +238,26 @@ def _apply_author_info_to_metadata_file(metadata_dict: dict, original_metadata_f
     return metadata_dict
 
 
+def _apply_components_py_file_to_metadata_file(
+    metadata_dict: dict,
+    components_py_file_path: Optional[Path] | None,
+) -> dict:
+    """Apply the components.py file to the metadata file before uploading it to GCS."""
+    if components_py_file_path:
+        metadata_dict = set_(
+            metadata_dict,
+            "data.generated.components_py.required",
+            True
+        )
+        metadata_dict = set_(
+            metadata_dict,
+            "data.generated.components_py.md5",
+            compute_gcs_md5(components_py_file_path),
+        )
```

Comment:
This `md5` would be a minimum checksum for now. During execution, the caller would check the `components.py` contents against the hash before executing.